### DISCUSSION TITLE
v1.0.3.1 difference between rp2040 and rp2040_big

### DISCUSSION THREAD

**Discussion by robzed** [tabemann/zeptoforth#63] [General]
There are two versions in the bin/1.0.3.1/ directory - rp2040 and rp2040_big.
What's the difference?

**Comment by tabemann** [comment]
The difference between the two is that `rp2040` provides 960 KiB of flash dictionary space, 1 MiB of block space (before taking block metadata into account), 48 KiB of flash dictionary index space and (for reasons due to how flash erasing works, 16 KiB of wasted flash space), while `rp2040_big` provides 1472 KiB of flash dictionary space, 512 MiB of block space (before taking block metadata into account), 60 KiB of flash dictionary index space and (for reasons due to how flash erasing works, 4 KiB of wasted flash space). Note that exactly one cell of the wasted flash space is used, to indicate how much of the flash in the dictionary is actually used; this will be erased the first time the RP2040 is booted after it is initially flashed. Note that the flash dictionary index space exists in both flash and RAM, so with an `rp2040` build it takes up 48 KiB of RAM, while with an `rp2040_big` build it takes up 60 KiB of RAM.

Of course, you are probably wondering about why this is necessary. This is that when zeptoIP and the CYW43439 firmware and driver are all installed, they take up a very large portion of the flash dictionary space and the flash dictionary index, necessitating expanding the flash dictionary space and the flash dictionary index to enable the user to actually be able to use flash once the are all installed.
